test/test.js

```diff
@@ -147,4 +147,74 @@ describe('Document', function(){
             });
         });
     });
+
+    describe('#findOneAndUpdate()', function(){
+        var sub_document = new TestModel({
+            testNumber: 424,
+            testArray: ['test3', 'test4'],
+            testObject: {
+                testObjectElement: 'something wicked this way comes'
+            },
+            ignoredField: 'Ignore me'
+        });
+
+        it('An historical record should be created when a document is saved.', function(done){
+            sub_document.save(function (e, obj) {
+                assert.equal(e, null);
+                assert.notEqual(obj, null);
+
+                sub_document = obj;
+
+                obj.historical(function (e, details) {
+                    assert.equal(e, null);
+                    assert.notEqual(details, null);
+
+                    var diff = _.merge(details.pop().diff, {_id: obj._id, __v: obj.__v});
+
+                    assert.strictEqual(diff.ignoredField, undefined);
+                    
+                    var withoutIgnored = obj.toObject();
+                    delete withoutIgnored['ignoredField'];
+                    
+                    assert.deepEqual(withoutIgnored, diff);
+                    assert.equal(obj.testString, 'My default value');
+                    
+                    done();
+                });
+            });
+        });
+
+        it('An historical record should be created when a document is modified.', function(done){
+
+            var update = {
+                $set: {
+                    testObject: {
+                        testObjectElement: 'this is a shift in concsiousness'
+                    }
+                }
+            }
+
+            var query = {
+                testObject: {
+                    testObjectElement: 'something wicked this way comes'
+                }
+            }
+
+            TestModel.findOneAndUpdate(query, update, function (e, obj) {
+                assert.equal(e, null);
+                assert.notEqual(obj, null);
+
+                sub_document = obj;
+
+                obj.historical(function (e, details) {
+                    assert.equal(e, null);
+                    assert.notEqual(details, null);
+
+                    assert.deepEqual(details.pop().diff, {testObject:{testObjectElement: 'something wicked this way comes'}});
```

Comment:
If the update was saved appropriately in the historical collection this value should equal `this is a shift in concsiousness`.